### ISSUE TITLE
Untap NPM module name

### ISSUE THREAD

**Issue by scottcorgan** [hville/untap#1]
@hville 

I know this is a bit of a stretch, but I am curious if you'd be willing to change the name of your module to `detap` on NPM (which, in my opinion, might be a name that still suits the use of the module.

The reason I ask is that I'd like to use the `untap` package name on npm for a set of tap formatters that I'm going to publish very soon.

Either way, I appreciate your time!

**Comment by hville** [comment]
Hi Scott,
I have no particular attachment to the name but got concerned over the implications of a change. There are many side effects and I am concerned about unintended consequences. I am open to discuss this but I have a strong feeling it is just bad practice to break things this way, even if the user base is very small. Can you elaborate on what you had in mind and how to implement it?
Cheers,
Hugo.

**Comment by scottcorgan** [comment]
That is a valid thought. 

I'm not usually an advocate of changing a module's purpose either, but since you're still at version 0.0.4, anything depending on the module will continue to work if using 0.0.4 or older. My thought was to use the package name and publish at 1.0. This will allow backwards compatibility with anything using untap now. NPM doesn't allow you to delete previous versions, so it's guarenteed.

My use case for this module is to create an ecosystem of TAP formatters and parsers as the future of my other moduls ([tap-spec](https://github.com/scottcorgan/tap-spec), [tap-dot](https://github.com/scottcorgan/tap-dot), [tap-out](https://github.com/scottcorgan/tap-out)).

Basically, you would just add me as an owner (`npm owner add untap scottcorgan`) on NPM and I could take it from there.

**Comment by hville** [comment]
The use case for untap was for a very lightweight summary that also shows skipped items. I only do local npm installs and often face crappy and/or metered internet connections when travelling. 

I've used tap-dot before but the 155 files and half a meg was too big for many small modules that get synced multiple times on different systems.

Replacing this with a full suit is too big of a change, even at 1.0.0, and blocks future updates that will need to be in a new name.

Thanks for reaching out though. I will stick to compound and/or name-spaced modules names in the future.

**Comment by scottcorgan** [comment]
Actually, the only thing I was asking is if you'd consider renaming your module to `detap` and letting me have the `untap` name on NPM. I didn't want to change your module in any way. Your code wouldn't change at all, just the name of the package in your package.json file.

Does that make sense?

**Comment by hville** [comment]
Yes it does make sense. Here is how I see it:
- **pro :** The popular suit that you are working on gets a short name that is dear to you  
- **con :** A relatively small set of users will get unexpected updates and will have to edit their JSON files once they understand what happened. I will also have to change all my public and private projects, leading to more changes.

All in all I still think it is bad practice to have such a break for a non-essential reason. For example, I would be pissed if updating `tt` would get me the full `tap` instead because of a name change.

**Comment by scottcorgan** [comment]
Fair enough. I appreciate the discussion.

**Comment by hville** [comment]
Hey, Scott. If you are still interested, I'll let it go. looks like I am
the only user and I am moving all my stuff to scoped @names/
Hugo

Hugo
+60 12 4997551 (KL)
